fundamentals-recap/fundamentals-vocabulary.md

```diff
@@ -0,0 +1,186 @@
+```
+
+Examples:
+
+```python
+print(type("This is a string!"))
+print(type(6789 * 100))
+print(type(3.3))
+print(type(["do", "re", "mi"]))
+
+my_bicycle_details = {
+    "frame_color": "red",
+    "num_of_wheels": 2,
+    "grip_tape_color": "turquoise"
+}
+print(type(my_bicycle_details))
+print(type(my_bicycle_details["num_of_wheels"]))
+```
+
+### Forcing a Data Type
+
+We can convert a value to a different data type.
+
+We can use these methods to convert a value into another data type
+
+- `int()`
+- `float()`
+- `str()`
+- `bool()`
+
+```python
+my_lucky_number = 7
+my_num_as_float = float(my_lucky_number)
+my_num_as_str = str(my_lucky_number)
+my_num_as_bool = bool(my_lucky_number)
+
+print(my_num_as_float)
+print(my_num_as_str)
+print(my_num_as_bool)
+
+my_lucky_fraction = 3.33
+my_fraction_as_int = int(my_lucky_fraction)
+
+print(my_fraction_as_int)
+```
+
+Callout: Often times, the behavior of converting isn't very predictable. What happens when we cast an empty list into a string? Or an integer into a boolean? It is best to experiment with converting in a REPL before using it in code.
+
+## Literals Are The Thing Itself
+
+In programming, a **literal** is a way to write a value that is the value itself. If the value _looks_ the same as its value, then it is a literal.
+
+```python
+"This string is a string literal, because this string is literally a string."
+```
+
+This is in contrast to variables, which have a value, but are not the value itself, nor are they written as the value itself.
+
+```python
+my_string = "Hello, world!"
+```
+
+The string `"Hello, world!"` and the variable `my_string` both have the same value. However, they differ in how the value is written. `"Hello, world!"` is a string literal because what we see is literally what the value is. `my_string` does not _look_ like what its value is. `my_string` is not _literally_ the string `"Hello, world!"`.
+
+```python
+my_dict = {
+    "bananas": 1,
+    "mangos": 2
+}
+```
+
+The dictionary `{"bananas": 1, "mangos": 2}` is a dictionary literal because it looks like its value. The variable `my_dict` refers to a dictionary as its value, but isn't a dictionary literal.
+
+## Printing and Formatting Strings
+
+Often times, we want to combine strings and non-string expressions.
+
+Python allows us to combine these by making **f-strings.** These strings have a prefix of `f` at the beginning, and evaluate any expression that is contained in `{}` curly braces.
+
+```python
+f"The string to format and include Python syntax, where expressions inside curly braces turn into a value!: {2 + 3}"
+```
+
+We can observe these examples:
+
+```python
+print(f"The sum of 2 + 3 is {2 + 3}")
+
+pineapples = "mY pInEaPpLe mEsSaGe"
+
+print(f"Here is {pineapples}!")
+```
+
+There are many other ways to combine strings and non-string values. Some methods include concatenation, or the `format()` method. Feel free to research these and use them; follow your curiosity!
```

Comment:
## Check for Understanding

<!--BEGIN CHALLENGE-->

### !challenge

* type: short-answer
* id: e255bf2d-f703-4a22-ba13-f142584bba93
* title: Expressions and Statements

##### !question

Using the code snippet below, on line 2, what is the expression?

```python

1   test_num = 5
2   if test_num > 10:
3       print "Bigger than ten!"

```
##### !end-question

##### !answer

test_num > 10

##### !end-answer

<!--optional-->
##### !hint

An expression is a piece of code that evaluates to a value.  In the conditional `if` statement, what is the portion that evaluates to True or False?

##### !end-hint

<!--optional-->
##### !explanation

In the `if` statement, `test_num > 10` will evaluate to False.  It is the expression that is being evaluated in the statement.

##### !end-explanation

### !end-challenge

<!--END CHALLENGE-->

<!--BEGIN CHALLENGE-->

### !challenge

* type: short-answer
* id: f07c693b-b2f2-4e16-a114-74cf6fcfe43a
* title: Literals

##### !question

Using the code snippet below, identify at least 1 literal:

```python

1   test_num = 5
2   if test_num > 10:
3       print "Bigger than ten!"

```

##### !end-question

##### !answer

/5|10|"Bigger than ten!"/

##### !end-answer

<!--optional-->
##### !hint

There are three literals in the code snippet.  Remember, the literal is the value written as itself.

##### !end-hint

<!--optional-->
##### !explanation

There are three literals in the code snippet, `5`, `10` and `"Bigger than ten!"`.

##### !end-explanation

### !end-challenge

<!--END CHALLENGE-->

Reply:
Added these manually